### ISSUE TITLE
Generated transaction IDs are not unique

### ISSUE THREAD

**Issue by OlegMazurov** [hashgraph/hedera-sdk-java#1557]
### Description

I generate many transactions in parallel from the same account (it's a setup stage in a benchmark built on top of the Java Client SDK; the transactions are `AccountCreateTransaction`). Occasionally, the same ID is generated for two different transactions. If those transactions are submitted to the same node I get the `DUPLICATE_TRANSACTION` error for the last one. If, however, they are submitted to different nodes, it becomes possible to get a receipt for a different transaction that is accepted as valid. In particular, `AccountCreateTransaction` receives a wrong account ID that can not be used with the key provided with the transaction.
The problem is with the `TransactionId.generate()` method:
```
    public static TransactionId generate(AccountId accountId) {
        Instant instant = Clock.systemUTC().instant().minusNanos((long) (Math.random() * 5000000000L + 8000000000L));
        return new TransactionId(accountId, instant);
    }
```
The randomization works OK at a low speed but at a high TPS (up to 25K in my case) the probability of generating the same value becomes significant. A strictly monotonic clock source is required here. Unfortunately, existing Java clock sources do not provide such guarantee in multithreaded environment. 

Thanks to the possibility of setting my own transaction ID, I have been able to implement the following scheme that works reliably in my benchmark:
```
    private final long adjustment = System.currentTimeMillis() * 1_000_000L - System.nanoTime();
    private final AtomicLong monotonicTime = new AtomicLong();

    private TransactionId generateTxnId(Client client) {
        long currentTime;
        long lastTime;
        do {
            currentTime = System.nanoTime() + adjustment;
            lastTime = monotonicTime.get();
            if (currentTime <= lastTime) currentTime = lastTime + 1;
        } while (!monotonicTime.compareAndSet(lastTime, currentTime));
        return new TransactionId(client.getOperatorAccountId(), Instant.ofEpochSecond(0, currentTime));
    }
```
Something along those lines should be implemented in Client SDKs.

### Steps to reproduce

I used [Load Generator](https://github.com/hashgraph/network-load-generator) to run `CryptoTransferLoadTest` against a 7-node network.

### Hedera network

other

### Version

v2.24.0

### Operating system

Linux

**Comment by thenswan** [comment]
Hi @OlegMazurov, and thank you for raising this issue. To give you an update -- it is currently in progress :)

**Comment by iron4548** [comment]
Unsure if this one is related but even since I installed 2.29.0 I've been running into this error:

Hedera transaction `0.0.xxxxx@xxxxxxxxxxxxxxxx.123000000` failed pre-check with the status `INVALID_TRANSACTION_START` with all kind of transactions (like ContractExecuteTransaction, TokenAssociateTransaction etc.) 

Note that there are only 3 decimal places in the nanosecond part, and the rest is zero. This seems unusual.

I rolled back to 2.28.0 and my code worked without issues.

**Comment by OlegMazurov** [comment]
See also https://github.com/hashgraph/hedera-services/pull/7959

**Comment by iron4548** [comment]
> See also [hashgraph/hedera-services#7959](https://github.com/hashgraph/hedera-services/pull/7959)

Thanks, I had a look but it's not clear on what I need to do. It appears to be resolved in v0.42 - is that for com.hedera.hashgraph:hedera-protobuf-java-api:0.42.0 ? I'm not including that one in my build.gradle. I was only using "com.hedera.hashgraph:sdk:2.29.0"
